Content.Server/Atmos/Piping/Trinary/EntitySystems/GasMixerSystem.cs

```diff
@@ -203,5 +204,38 @@ private void OnOutputPressureChangeMessage(EntityUid uid, GasMixerComponent mixe
                 $"{EntityManager.ToPrettyString(args.Session.AttachedEntity!.Value):player} set the ratio on {EntityManager.ToPrettyString(uid):device} to {mixer.InletOneConcentration}:{mixer.InletTwoConcentration}");
             DirtyUI(uid, mixer);
         }
+
+        /// <summary>
+        /// Returns the gas mixture for the gas analyzer
+        /// </summary>
+        private void OnMixerAnalyzed(EntityUid uid, GasMixerComponent component, GasAnalyzerScanEvent args)
+        {
+            if (!EntityManager.TryGetComponent(uid, out NodeContainerComponent? nodeContainer))
+                return;
+
+            var gasMixDict = new Dictionary<string, GasMixture?>();
+
+            nodeContainer.TryGetNode(component.InletOneName, out PipeNode? inletOne);
+            nodeContainer.TryGetNode(component.InletTwoName, out PipeNode? inletTwo);
+
+            if(inletOne != null)
+                gasMixDict.Add($"{inletOne.CurrentPipeDirection} {Loc.GetString("gas-analyzer-window-text-inlet")}", inletOne.Air);
+            if(inletTwo != null)
+                gasMixDict.Add($"{inletTwo.CurrentPipeDirection} {Loc.GetString("gas-analyzer-window-text-inlet")}", inletTwo.Air);
+            if(nodeContainer.TryGetNode(component.OutletName, out PipeNode? outlet))
+                gasMixDict.Add(Loc.GetString("gas-analyzer-window-text-outlet"), outlet.Air);
+
+            /*
+            bool flipped;
+            if (inletOne != null && inletTwo != null)
+            {
+                flipped = inletOne.CurrentPipeDirection.ToDirection() == inletTwo.CurrentPipeDirection.ToDirection().GetClockwise90Degrees();
+            }
+            else
+                flipped = false;
+            */
+            args.GasMixtures = gasMixDict;
+            args.DeviceFlipped = inletOne != null && inletTwo != null && inletOne.CurrentPipeDirection.ToDirection() == inletTwo.CurrentPipeDirection.ToDirection().GetClockwise90Degrees();
```

Comment:
i definitely feel like theres still a better way to do this but eh its good

Reply:
probably, but it's the best I could come up with after doodling it all out. It's an arbitrary convention to fit the UI for sure

open to suggestions if someone thinks of a better way